modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rules/editor/Editor.es.js

```diff
@@ -371,19 +371,30 @@ describe('Editor', () => {
 							fireEvent.click(getByText('value'));
 						});
 
-						await waitFor(() => {
-							document
-								.querySelectorAll('.timeline-item')[1]
-								.querySelectorAll('.ddm-field')[3]
-								.querySelector(selector);
-						});
+						if (type === 'rich_text') {
+							await waitFor(() => {
+								document.querySelectorAll(selector);
+							});
 
-						expect(
-							document
-								.querySelectorAll('.timeline-item')[1]
-								.querySelectorAll('.ddm-field')[3]
-								.querySelector(selector)
-						).toBeTruthy();
+							expect(
+								document.querySelectorAll(selector)
+							).toBeTruthy();
+						}
+						else {
+							await waitFor(() => {
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector(selector);
+							});
+
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector(selector)
+							).toBeTruthy();
```

Comment:
Hey @rebsilva, you can do the `expect` inside of the `waitFor`, avoiding code repetition this way:

```js
await waitFor(() => {
	const richTextElement = document
		.querySelectorAll('.timeline-item')[1]
		.querySelectorAll('.ddm-field')[3]
		.querySelector(selector);
	
	expect(richTextElement).toBeTruthy();
});
```

Reply:
does it have the same behavior since the `waitFor` won't be sure to finish before the `expect(richTextElement).toBeTruthy()` because they are now inside the same function?

Reply:
If the `waitFor` completes successfully, it guarantees that the condition inside has been met. The `expect` statement within the `waitFor` will only be evaluated once the condition is true. This way, the behavior remains the same, ensuring that the `expect` is not executed until the `waitFor` has finished.

Reply:
i see! thanks!